### ISSUE TITLE
CKAsset and CKReference support?

### ISSUE THREAD

**Issue by brendand** [joulupukki/BTCloudKitSync#14]
Hi,

This looks like a great project. Just what I was looking for. However, how does it handle CKAsset and CKReference objects?

I need to be able to store a variable amount of attachments on a record and I thought maybe creating multiple child CKRecords and using CKReference to link them back to the parent CKRecord might be a way of doing that, each related record having its own CKAsset object on it.

Is there any planned support for that?

Looking at the code I wonder if it would be better to allow the delegate to provide the CKRecord objects to sync rather than providing an array of NSDictionary objects? If it was that way, then I could provide an array of CKRecord objects which already had their relationships to other CKRecord objects established and already populated with their CKAsset fields.

Thanks,

Brendan

**Comment by brendand** [comment]
Is this project still in active development?

**Comment by joulupukki** [comment]
I'm still using it in an active app on the App Store, but haven't touched this particular code in a while.

My original thinking in the implementation was to allow implementors of the protocol to not have to know anything about the CKRecord or CK*** objects at all. I wonder if it would work if there was an optional key/value pair in the returned dictionary that was a CKRecord and if so, the sync implementation could treat it differently by looking for CKReferences? Do you have time to look at an implementation like this? Or maybe propose a solution based on what you suggested?

**Comment by brendand** [comment]
Well the nice thing if it worked with CKRecords instead of dictionaries is the CKRecord can be created by the app however it wants to and your framework would just worry about syncing those rather than having to translate between dictionaries and CKRecords. A CKRecord really behaves kind of a like a smarter NSDictionary, so just going directly to a CKRecord from my own model objects makes much more sense than going from my model objects to a dictionary and then you translating from the dictionary to the CKRecord.

**Comment by joulupukki** [comment]
Makes sense. I don't currently have time to spend testing this out, but would be open to help if you'd want to modify and test it out.
